programs/x509/cert_req.c

```diff
@@ -145,11 +164,13 @@ int main(int argc, char *argv[])
     mbedtls_pk_context key;
     char buf[1024];
     int i;
-    char *p, *q, *r;
+    char *p, *q, *r, *r2;
     mbedtls_x509write_csr req;
     mbedtls_entropy_context entropy;
     mbedtls_ctr_drbg_context ctr_drbg;
     const char *pers = "csr example app";
+    mbedtls_x509_san_list *cur, *prev;
```

Comment:
All of the newly added variables are local to the "san" case. I would suggest moving them there.

Reply:
This comment still stands.

Reply:
I'm not sure about this. What is wrong with these variables declaration here?

Where do you suggest to move them (line 222)?

Reply:
We're putting things in a scope that's too big, and it introduces unnecessary mess. It's a minor, if you don't want to, that's not a problem, but It's just cleaner to keep things declared as close to where they are used.

Reply:
Ok, but `cur` and `prev` are also used at the end to release memory after `exit` label. So the scope is correct. I can move `ip`.